conans/client/tools/intel.py

```diff
@@ -0,0 +1,107 @@
+import os
+import platform
+from contextlib import contextmanager
+
+from conans.client.tools.env import environment_append, env_diff
+from conans.client.tools.win import is_win64, _system_registry_key, MSVS_YEAR
+from conans.errors import ConanException
+from conans.util.env_reader import get_env
+
+
+# https://software.intel.com/en-us/articles/intel-compiler-and-composer-update-version-numbers-to-compiler-version-number-mapping
+INTEL_YEAR = {"19.1": "2020",
+              "19": "2019",
+              "18": "2018",
+              "17": "2017",
+              "16": "2016",
+              "15": "2015"}
+
+
+def intel_installation_path(version, arch):
+    installation_path = get_env("CONAN_INTEL_INSTALLATION_PATH")
+    if installation_path:
+        return installation_path
+
+    system = platform.system()
+    if system in ["Linux", "Darwin"]:
+        subdir = "mac" if system == "Darwin" else "linux"
+        year = INTEL_YEAR.get(version)
+        installation_path = os.path.join(os.sep, "opt", "intel",
+                                         "compilers_and_libraries_%s" % year, subdir)
+    elif system == "Windows":
+        if arch == "x86":
+            intel_arch = "IA32"
+        elif arch == "x86_64":
+            intel_arch = "EM64T"
+        else:
+            raise ConanException("don't know how to find Intel compiler on %s" % arch)
+        if is_win64():
+            base = r"SOFTWARE\WOW6432Node\Intel\Suites\{version}".format(version=version)
+        else:
+            base = r"SOFTWARE\Intel\Suites\{version}".format(version=version)
```

Comment:
At least for the 2019 compiler, the version needs to be `19.0` not `19`. This is likely true of earlier compilers, not sure about the 2020 compiler.

Reply:
thanks, I've corrected to add .0, if needed